atlp-rwanda/rca-phantom-team1-bn
# CRUD locations

# Location CRUD

## Description

This pull request implements the functionality to create, update, and delete locations on the back-end using nodejs-express-postgresql.

## Changes

- Added a new file `location.js` in the `routes` folder that contains the endpoints for location CRUD operations
- Added a new file `locationController.js` in the `controllers` folder that contains the logic for handling location requests
- Added a new file `locationModel.js` in the `models` folder that contains the schema and queries for location data
- Added a new file `location.test.js` in the `tests` folder that contains the unit tests for location endpoints


## How to test

- Run `npm install` to install the dependencies
- Run `npm run migrate` to create the locations table in the database
- Run `npm run test` to run the unit tests
- Run `npm start` to start the server
- Use Postman or curl to test the following endpoints:

  - POST `/api/locations` to create a new location
    - Request body should have the following fields: name (string), address (string), latitude (number), longitude (number), status (string)
    - Response should have a status code of 201 and a JSON object with the created location data
  - GET `/api/locations` to get all locations
    - Response should have a status code of 200 and a JSON array with all location data
  - GET `/api/locations/:id` to get a specific location by id
    - Response should have a status code of 200 and a JSON object with the requested location data
    - If the id is not valid or not found, response should have a status code of 404 and an error message
  - PUT `/api/locations/:id` to update a specific location by id
    - Request body should have any of the following fields: name (string), address (string), latitude (number), longitude (number), status (string)
    - Response should have a status code of 200 and a JSON object with the updated location data
    - If the id is not valid or not found, response should have a status code of 404 and an error message
  - DELETE `/api/locations/:id` to delete a specific location by id
    - Response should have a status code of 200 and a JSON object with the deleted location data
    - If the id is not valid or not found, response should have a status code of 404 and an error message

## Screenshots

![image](https://user-images.githubusercontent.com/68426990/229793758-9f99ae50-6d04-4e15-8921-838acd5ec1e6.png)


## Checklist

- [x] I have followed the coding standards and style guide
- [x] I have written clear and concise commit messages
- [x] I have tested my code and fixed any bugs or errors
- [x] I have added relevant documentation and comments
